sdk/python/feast/infra/online_stores/contrib/hazelcast_online_store/hazelcast_online_store.py

```diff
@@ -0,0 +1,252 @@
+            for feature_name, value in values.items():
+                feature_value = value.SerializeToString().hex()
+                __key = str(entity_key_bin) + feature_name
+                fv_map.put(
+                    __key,
+                    HazelcastJsonValue(
+                        {
+                            "entity_key": entity_key_bin,
+                            "feature_name": feature_name,
+                            "feature_value": feature_value,
+                            "event_ts": event_ts_utc,
+                            "created_ts": created_ts_utc,
+                        }
+                    ),
+                    config.online_store.key_ttl_seconds,
+                )
+                if progress:
+                    progress(1)
+
+    def online_read(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        entity_keys: List[EntityKeyProto],
+        requested_features: Optional[List[str]] = None,
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+        online_store_config = config.online_store
+        assert isinstance(online_store_config, HazelcastOnlineStoreConfig)
+
+        client = self._get_client(online_store_config)
+        entries: List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]] = []
+        fv_map = client.get_map(_table_id(config.project, table))
+
+        hz_keys = []
+        entity_keys_str = {}
+        for entity_key in entity_keys:
+            feature_keys = []
+            entity_key_str = str(
+                serialize_entity_key(
+                    entity_key,
+                    entity_key_serialization_version=config.entity_key_serialization_version,
+                ).hex()
+            )
+            if requested_features:
+                for feature in requested_features:
+                    feature_keys.append(entity_key_str + feature)
+            else:
+                for f in table.features:
+                    feature_keys.append(entity_key_str + f.name)
+            hz_keys.extend(feature_keys)
+            entity_keys_str[entity_key_str] = feature_keys
+
+        data = fv_map.get_all(hz_keys).result()
+        entities = []
+        for _key in hz_keys:
+            try:
+                data[_key] = data[_key].loads()
+                entities.append(data[_key]["entity_key"])
+            except KeyError:
+                continue
+
+        for curr_key in entity_keys_str:
+            if curr_key in entities:
+                entry = dict()
+                event_ts: Optional[datetime] = None
+                for _key in entity_keys_str[curr_key]:
+                    row = data[_key]
+                    value = ValueProto()
+                    value.ParseFromString(bytes.fromhex(row["feature_value"]))
+                    entry[row["feature_name"]] = value
+                    event_ts = datetime.utcfromtimestamp(row["event_ts"])
```

Comment:
Instead of `utcfromtimestamp` the following should be used:
```
datetime.fromtimestamp(row["event_ts"], tz=timezone.utc)
```
See: https://blog.ganssle.io/articles/2019/11/utcnow.html

Reply:
Thanks for this info, fixed now: https://github.com/mehmettokgoz/feast/pull/1/commits/9f8174c72faf2bbdbd6f7fe4b414d63f20338600#diff-47764bf85c9631f63e542e314d1d29579a85e570b2813040caaade67743322d7R237.